src/coreclr/vm/unsafeaccessors.cpp

```diff
@@ -75,16 +93,258 @@ namespace
 
         UnsafeAccessorKind Kind;
         MethodDesc* Declaration;
-        MetaSig DeclarationSig;
+        SigPointer DeclarationSig;  // This is the official declaration signature. It may be modified
+                                    // to include the UnsafeAccessorTypeAttribute types.
+        MetaSig DeclarationMetaSig;
+        NewArrayHolder<ParamDetails> TranslatedParams; // Redefined types from UnsafeAccessorTypeAttribute usage.
+                                                       // Return type is at 0 index. Function arguments are 1 to N.
         SigPointer TargetTypeSig;
         TypeHandle TargetType;
         bool IsTargetStatic;
         MethodDesc* TargetMethod;
         FieldDesc* TargetField;
     };
 
+    void AppendTypeToSignature(
+        SigBuilder& sig,
+        TypeHandle th,
+        Instantiation inst)
```

Comment:
Does this actually need `inst` argument? Can `inst` be fetched from `th` inside the method?

Reply:
I can do that. Not sure it is clearer, but I can.